experimental/ixdtf/src/parser.rs

```diff
@@ -61,314 +64,241 @@ struct ParsedDateTime {
 }
 
 #[derive(Debug, Clone, Copy, PartialEq)]
-struct DateTimeParser {}
-
-impl DateTimeParser {
-    fn parse_date_extended_year<'a, I>(
-        iter: &mut core::iter::Peekable<I>,
-    ) -> Result<Option<i32>, ParseError>
-    where
-        I: Iterator<Item = &'a u8>,
-    {
-        let mut year: i32 = 0;
-        // TODO: value assigned to `sign` is never read
-        #[allow(unused_assignments)]
-        let mut sign = 1;
-        if iter.peek() == Some(&&b'+') {
-            sign = 1;
-        } else if iter.peek() == Some(&&b'-') {
-            sign = -1
-        } else {
-            return Err(ParseError::DateExtendedYear);
-        }
-        iter.next();
-        let mut cnt = 0;
-        while cnt < 6 {
-            if let Some(y) = iter.peek() {
-                if **y >= b'0' && **y <= b'9' {
-                    year = year * 10 + **y as i32 - b'0' as i32;
+struct DateTimeParser<'a> {
+    bytes: &'a [u8],
+}
+
+impl<'a> DateTimeParser<'a> {
+    fn parse_date_extended_year(&mut self) -> Result<Option<i32>, ParseError> {
```

Comment:
nit: in the long run all of these should be documented, saying what they're parsing with examples (and explaining things like why there's an Option here)

for now it's fine to punt docs till later

Reply:
add simple comments for the structs in this file. Will add more once the code is stable